lib/beacon/content/component_definition.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Beacon.Content.ComponentDefinition do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Beacon.Content.ComponentDefinition
+
+  @type t :: %__MODULE__{}
+
+  @primary_key {:id, :binary_id, autogenerate: true}
+  @foreign_key_type :binary_id
+  schema "beacon_component_definitions" do
+    field :name, :string
+    field :thumbnail, :string
+    field :blueprint, :map
+    belongs_to :component_category, ComponentCategory
```

Comment:
We don't need `Beacon.ComponentCategories` **if** the only attr is `name` and we won't expose that to users create custom categories. It can be a [enum](https://hexdocs.pm/ecto/Ecto.Enum.html) instead:

```suggestion
    field : component_category, Ecto.Enum, values: [:header, :sign_in, :sign_up, :stats, :footer, :basic]
```

Reply:
Users will not create their own categories, I like the idea of removing one table